src/simple_rules.ml

```diff
@@ -19,7 +19,20 @@ let user_rule sctx ?extra_bindings ~dir ~scope (rule : Rule.t) =
   let targets : SC.Action.targets =
     match rule.targets with
     | Infer -> Infer
-    | Static fns -> Static (List.map fns ~f:(Path.relative dir))
+    | Static fns ->
+      let f fn =
+        let loc = String_with_vars.loc fn in
+        match SC.expand_vars_string sctx ~scope ~dir fn with
+        | "." | ".." ->
+          Errors.fail loc "'.' and '..' are not valid filenames"
+        | fn ->
+          if Filename.dirname fn <> Filename.current_dir_name then
+            Errors.fail
+              loc
+              "%s does not denote a file in the current directory" fn;
+          Path.relative ~error_loc:loc dir fn
+      in
+      Static (List.map fns ~f)
```

Comment:
One unfortunate thing about this code is that it won't be able to extend variables into multiple targets. That's a bit unnatural, especially given how deps work. I suggest that we use:

```
SC.expand_vars ~mode:String_with_vars.Mode.Many
```

do the error checking like you've done, and then use `Value.to_path`.

Reply:
I will look into it. But just to make sure I understand: currently there are no "list" variables which are static enough to be used in the `targets` field, right?

Reply:
Technically, there are a few. Things like flags or binareis defined in `Ocaml_config.t`. Not the most useful things, but why not keep things consistent.